Serializing "complex" field returns None instead of the value since 3.4

(Not a great title — I hope the description will clarify.)

## Steps to reproduce

First: `pip install django-phonenumber-field`

With the following model and serializer:

```python
from django.db.models import Model
from phonenumber_field.modelfields import PhoneNumberField
from rest_framework.serializers import ModelSerializer

class PhoneModel(Model):
    number = PhoneNumberField()

class PhoneSerializer(ModelSerializer):
    class Meta:
        model = PhoneModel
        fields = ['number']
```

## Expected behavior

This test used to pass (until version 3.3.3):

```python
def test_phone_serializer():
    phone = models.PhoneModel(number='+33610293847')
    data = models.PhoneSerializer(phone).data
    assert data['number'] == '+33610293847'
```

## Actual behavior

The test fails (since version 3.4) with:

```
    def test_phone_serializer():
        phone = models.PhoneModel(number='+33610293847')
        data = models.PhoneSerializer(phone).data
>       assert data['number'] == '+33610293847'
E       assert None == '+33610293847'
```

## Analysis

I bisected this regression to 9c996d7d2aa4137c8ba29afa2253efec8d6db74f.

As far as I can tell, DRF used to get the string representation of the field and no longer does.

```
(Pdb) phone.number
PhoneNumber(country_code=33, national_number=610293847, extension=None, italian_leading_zero=None, number_of_leading_zeros=None, country_code_source=1, preferred_domestic_carrier_code='')
(Pdb) str(phone.number)
'+33610293847'
```

--- a/rest_framework/compat.py
+++ b/rest_framework/compat.py
@@ -125,7 +125,7 @@ def get_related_model(field):
 def value_from_object(field, obj):
     if django.VERSION < (1, 9):
         return field._get_val_from_obj(obj)
-    field.value_from_object(obj)
+    return field.value_from_object(obj)
 
 
 # contrib.postgres only supported from 1.8 onwards.

